### DISCUSSION TITLE
Import problem 500 Internal Server Error

### DISCUSSION THREAD

**Discussion by AndersonOuverney** [firefly-iii/firefly-iii#5582] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Good morning,
I'm finding it very difficult to import using the import tool located at this URL:
https://data-importer.firefly-iii.org/token

I get the following error:

error :(

 The server you tried to connect to responsed with an error.

Server error: `POST https://xxxxx.xxxxxxx.xxxxxxxx/oauth/token` resulted in a `500 Internal Server Error` response:  <!DOCTYPE html> <html lang="en"> (truncated...) 

My FireFly version is 5.6.8, do you have any tips I can do to solve the problem?

### Expected behaviour

I would like to be successful in importing transactions from my bank accounts as I have always managed to do without major difficulties.

### Debug information

 The server you tried to connect to responsed with an error.

Server error: `POST https://xxxxx.xxxxxxx.xxxxxxxx/oauth/token` resulted in a `500 Internal Server Error` response:  <!DOCTYPE html> <html lang="en"> (truncated...)

**Comment by JC5** [comment]
I see a few errors in the logs on my side:

- `Maximum execution time of 30 seconds exceeded`. I've fixed this on my end.

I can't see what the 500 error is, that can be found in your log files. Could you share that with me?

**Comment by AndersonOuverney** [comment]
Good morning,
Thanks for your answer, I don't know exactly which log you need so I'm uploading the 2 logs that changed today
[ff3-audit-2022-01-21.log](https://github.com/firefly-iii/firefly-iii/files/7914024/ff3-audit-2022-01-21.log)
[ff3-litespeed-2022-01-21.log](https://github.com/firefly-iii/firefly-iii/files/7914025/ff3-litespeed-2022-01-21.log)
.

**Comment by JC5** [comment]
I see an openSSL error but it's unfamiliar to me:

```
It was not possible to parse your key, reason
```

How did you install Firefly III? Did you manage to create an access token?

**Comment by AndersonOuverney** [comment]
I've had firefly installed for a long time, it's installed on a shared server, running over litespeed, it's always worked great.

I spent a few months without doing any work on the system, when I went to carry out the imports I saw that I had to perform the update.

I followed the upgrade steps in the guides, after several trials and errors I managed to upgrade to this current version.

Yes I have an access token created, including the importer appears in authorized applications.

I really don't know what to do to get firefly to work perfectly.

**Reply by JC5** [reply]
Me neither, I've never seen this error before. 

Try running `php artisan passport:keys --force` and then regenerate your token.

**Comment by AndersonOuverney** [comment]
I noticed that the URL
  https://data-importer.firefly-iii.org/callback
is having a problem with the certificate, is that the problem?

NET::ERR_CERT_AUTHORITY_INVALID
Subject: CloudFlare Origin Certificate

Issuer: The original certificate provided by the server is untrusted

Expires on: 23 de dez. de 2036

Current date: 21 de jan. de 2022

PEM encoded chain:
-----BEGIN CERTIFICATE-----
MIIF4DCCA8igAwIBAgIQaKS8V9IAbRPI2MJQkVZddDANBgkqhkiG9w0BAQsFADBH
MUUwQwYDVQQDDDxUaGUgb3JpZ2luYWwgY2VydGlmaWNhdGUgcHJvdmlkZWQgYnkg
dGhlIHNlcnZlciBpcyB1bnRydXN0ZWQwHhcNMjExMjI3MDk1MjAwWhcNMzYxMjIz
MDk1MjAwWjBiMRkwFwYDVQQKExBDbG91ZEZsYXJlLCBJbmMuMR0wGwYDVQQLExRD
bG91ZEZsYXJlIE9yaWdpbiBDQTEmMCQGA1UEAxMdQ2xvdWRGbGFyZSBPcmlnaW4g
Q2VydGlmaWNhdGUwggIiMA0GCSqGSIb3DQEBAQUAA4ICDwAwggIKAoICAQDZSgNO
tX3tcDq9xo33kTRZDyfy/5hkJ5Q5rY5mC3TJIuXje7Xlxtot2VKgWTgaNl4UX3fw
lD9uNt9FYutwZ+yfcyCrfAdId4bifrVZLeEiUYLXTRpj5gD8AqUmE+xiPDKF/slg
LlbL3P/H+xFrRcrCTFPDjEJVnTB5SoUiVz4rnmSK0DUj/usdTFoVnvGg/ThyTeCG
KefsBOwX4uZ4vH1nc0XoTr6pAZ/2FWB5rbbydS7Iag7IyjCm++goaDTRGb0MjFmP
cOPp+JRNWIC5BP/a35vDvtQOz3kJA7BTWJV4gxdcj5T3b1Gk6CQDKyhABdONlmt4
ZXF21ubKufq6RB9ZfOwu2eo1rlweT2XJyF525POITJD1sqoSMQ2nUWdB/v4X9tK1
MCgUdpVNWSzUsCt5dSA4kQ6/EHyAxO2YHLfNF6E06z9rVJ7gQf5+SsOEVh2Y1/NO
cCZkJr4D1JoLND+aMQ0vJ31Q+9EQWlB9MkRD2aTlWFE1cdY+NuJNfBIFCaJ1yEUm
thMZBI2PuiER8hvGESb5SNit4ygmtf28nYm9Cd6m3xd5tlD65W9pd22Q5o2MMoSe
iS1TaZSmLr9RHeCYeoez59IMp4ezwXEsxmjGOsirbBVO7BZjMYyZs3Soxgz9rEmn
wErt9V6mUBHBkf9KqzrxJuz7vO/GIo0nYrUxlwIDAQABo4GsMIGpMAsGA1UdDwQE
AwIFoDAdBgNVHQ4EFgQU9YNoX0dshYhssF9meMF+S4HG2wMwHwYDVR0jBBgwFoAU
9YNoX0dshYhssF9meMF+S4HG2wMwHQYDVR0lBBYwFAYIKwYBBQUHAwIGCCsGAQUF
BwMBMAwGA1UdEwEB/wQCMAAwLQYDVR0RBCYwJIIRKi5maXJlZmx5LWlpaS5vcmeC
D2ZpcmVmbHktaWlpLm9yZzANBgkqhkiG9w0BAQsFAAOCAgEAUljAYd8A2RUtdp68
7pu5JdP4qU3wzLrSsfXCx2hcjr9cbZZ4BCrrCZ2NzMXsEsP866JPRM7y445H8nsl
fKmotmdpArwCNgdfpxFRhstAhsFyAqlhPTTJRFm4qeNNq5O5qXfsL1mo7jLH0wQb
7ZHRL9O9VO/vAZFP16j4Xe3egHbZI471cktnJEG5ZccOuzOhoSN9knjKC7zBdY+u
pL6kY7dCDswUO6yrp+uHERzJT9OYPM3Pb0x6kis1NGQW0xiDyVERUm8IATv1J/yG
gmjXkDi0IxoWaNo0YymDshKkHsi8MTs4xdK8KKipSZ2ZwnwRTZ1XhW4Xz8dZ/RlR
eYK2Vc8NwGwdnxJXpzEmdZ9E+tz5vOG4rDsopVJ/VSb7Dm4GrmNpqiYTGbsUQPXR
2MLkOAvWRFqOlYXxgMW9TgS+12sOEF/LutCdi16W211J+asP9yvqD6jLtP2F+MNM
QOxPGLwbdhWVJYd5DiyAWjIM/Kx18KUGiCqm2qel3aED5Rp8gbFMJrmxgT4iRWr/
6Bg2G4ixS+Uzn0dPdL3EiA6N0UTG+GRb7r8ZiBIGPro5YxUSGh7Erwa/Pg5w+BoQ
2yo4LVYHRJxef37MPXiHXqamQy/vG6i3kaoYvwNb8DWiPIgA3bOskAv8MIqYtYqV
fd4q+PlRjDeIhpRSIS1O+S3DNV8=
-----END CERTIFICATE-----

**Reply by JC5** [reply]
Yes, thanks! Should be fixed now.

**Comment by AndersonOuverney** [comment]
Thanks for the answers.

I just ran the process again, but I'm still seeing the same error code :(

**Reply by JC5** [reply]
Sorry about that. Ultimately, the root cause is that your public/private key pair (should be stored in `/storage`, `oauth-private.key` and `oauth-public.key`) is invalid.

I can't fix that for you, except suggest that you regenerate it with the `php artisan passport:keys --force` command.

To double validate this process, delete the files first, then regenerate the keys, *then* generate a new token and see if it works.

If that doesn't help, the problem could reside in your local installation, for example an old php version or library. But at this point I am totally out of my depth, sorry.

**Comment by AndersonOuverney** [comment]
Hi thanks for the reply.
I inform you that I deleted the files and ran the indicated command.
Nothing happened when doing this.

I'm thinking, do we have a problem with artisan?
I didn't get any feedback at the prompt when I ran the command.

**Reply by JC5** [reply]
The expected response is:

```
php artisan passport:keys --force
Encryption keys generated successfully.
```
